singleheader/amalgamate.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python3
+
+import sys
+import os.path
+import subprocess
+from pathlib import Path
+import re
+
+SCRIPTPATH=os.path.dirname(os.path.abspath(sys.argv[0]))
+PROJECTPATH=os.path.dirname(SCRIPTPATH)
+
+print(f"SCRIPTPATH={SCRIPTPATH} PROJECTPATH={PROJECTPATH}")
+
+
+print("We are about to amalgamate all simdjson files into one source file.")
+print("See https://www.sqlite.org/amalgamation.html and https://en.wikipedia.org/wiki/Single_Compilation_Unit for rationale.")
+if "AMALGAMATE_SOURCE_PATH" not in os.environ:
+  AMALGAMATE_SOURCE_PATH=os.path.join(PROJECTPATH,"src")
+else:
+  AMALGAMATE_SOURCE_PATH=os.environ["AMALGAMATE_SOURCE_PATH"]
+if "AMALGAMATE_INCLUDE_PATH" not in os.environ:
+  AMALGAMATE_INCLUDE_PATH=os.path.join(PROJECTPATH,"include")
+else:
+  AMALGAMATE_INCLUDE_PATH=os.environ["AMALGAMATE_INCLUDE_PATH"]
+if "AMALGAMATE_OUTPUT_PATH" not in os.environ:
+  AMALGAMATE_OUTPUT_PATH=os.path.join(SCRIPTPATH)
+else:
+  AMALGAMATE_OUTPUT_PATH=os.environ["AMALGAMATE_OUTPUT_PATH"]
+
+# this list excludes the "src/generic headers"
+ALLCFILES=["simdjson.cpp"]
+
+# order matters
+ALLCHEADERS=["simdjson.h"]
+
+found_includes=[]
+
+
+def doinclude(fid,file,line):
+  p=Path(AMALGAMATE_INCLUDE_PATH)/file
```

Comment:
I'm just nit picking, but I'd suggest using either one of os.path.*, or Pathlib, not both. The first 1/3 of this file uses os.path, the middle uses Pathlib, and the last 3rd uses os.join again.

Reply:
Don't worry about nitpicking, I appreciate it. I fixed this.